blockchain/fullblocktests/generate.go

```diff
@@ -1931,6 +1950,151 @@ func Generate(includeLargeReorg bool) (tests [][]TestInstance, err error) {
 	})
 	rejected(blockchain.ErrScriptMalformed)
 
+	// Create block that spends immature change from one ticket purchase in
+	// another ticket purchase.
+	//
+	//   ... -> brs3(14)
+	//                  \-> bmf25(15)
+	g.SetTip("brs3")
+	g.NextBlock("bmf25", outs[16], ticketOuts[16], func(b *wire.MsgBlock) {
+		prevBlock := g.Tip()
+		sstxIn := b.STransactions[6].TxIn[0]
+		sstxIn.ValueIn = prevBlock.STransactions[0].TxOut[2].Value
+		sstxIn.BlockIndex = 0
+		sstxIn.BlockHeight = prevBlock.Header.Height
+		sstxIn.PreviousOutPoint = wire.OutPoint{
+			Hash:  prevBlock.STransactions[0].TxHash(),
+			Index: 2,
+			Tree:  wire.TxTreeStake,
+		}
+	})
+	rejected(blockchain.ErrImmatureSpend)
+
+	// Create block with an invalid stake transaction signature script.
+	//
+	//   ... -> brs3(14)
+	//                  \-> bmf26(15)
+	g.SetTip("brs3")
+	g.NextBlock("bmf26", outs[15], ticketOuts[15], func(b *wire.MsgBlock) {
+		b.STransactions[5].TxIn[0].SignatureScript = repeatOpcode(txscript.OP_0, 12)
```

Comment:
Why 12?  I'd suggest creating a variable `var invalidP2SHRedeemScript = []byte{0x01, txscript.OP_FALSE}` next to the definition for `opTrueScript` and using it both here and the next test.  That way it is a valid script that evaluates to false those making it unspendable.